src/game/g_stats.c

```diff
@@ -574,6 +574,98 @@ void G_AddKillSkillPointsForDestruction(gentity_t *attacker, meansOfDeath_t mod,
 	CalculateRanks();
 }
 
+#define MAX_PLAYERS_ASSIST_TO_REWARDS 4
+#define MAX_ASSIST_ELAPSED_TIME 1500
+
+/**
+ * @brief G_SortPlayersDamageGiver
+ * @param[in] a
+ * @param[in] b
+ * @return An array sorted by higher damage received
+ */
+static int QDECL G_SortPlayersDamageGiver(const void *a, const void *b)
+{
+	const damageReceivedStats_t *ca = *(const damageReceivedStats_t **)a;
+	const damageReceivedStats_t *cb = *(const damageReceivedStats_t **)b;
+
+	if (ca->damageReceived > cb->damageReceived)
+	{
+		return -1;
+	}
+
+	if (cb->damageReceived > ca->damageReceived)
+	{
+		return 1;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief G_AddKillAssistPoints
+ * @param[in] target
+ * @param[in] attacker
+ */
+void G_AddKillAssistPoints(gentity_t *target, gentity_t *attacker)
+{
+	unsigned int          i;
+	unsigned              rewardedPlayers;
+	qboolean              complaintSent = qfalse;
+	damageReceivedStats_t *dmgReceivedSts[MAX_CLIENTS];
+
+	if (!target || !target->client)
+	{
+		return;
+	}
+
+	for (i = 0; i < ARRAY_LEN(target->client->dmgReceivedSts); ++i)
+	{
+		dmgReceivedSts[i] = &(target->client->dmgReceivedSts[i]);
+	}
+
+	qsort(dmgReceivedSts, ARRAY_LEN(target->client->dmgReceivedSts), sizeof(damageReceivedStats_t *), G_SortPlayersDamageGiver);
+
+	// rewards assist points
+	for (i = 0, rewardedPlayers = 0; i < ARRAY_LEN(dmgReceivedSts) && rewardedPlayers < MAX_PLAYERS_ASSIST_TO_REWARDS; ++i)
+	{
+		gentity_t *ent;
+
+		// no more damage received, no more damage to parse
+		if (dmgReceivedSts[i]->damageReceived <= 0)
+		{
+			break;
+		}
+
+		// skip old damage received
+		if (dmgReceivedSts[i]->lastHitTime + 1500 < level.time)
+		{
+			continue;
+		}
+
+		ent = &g_entities[dmgReceivedSts[i] - target->client->dmgReceivedSts];
+
+		// ignore damage received from self and attacker (killer)
+		if (ent == target || ent == attacker)
+		{
+			continue;
+		}
+
+		// reward only opposite team
+		if (ent->client->ps.teamNum != target->client->ps.teamNum)
```

Comment:
This should be `sess.sessionTeam` for safety
```suggestion
		if (ent->client->sess.sessionTeam != target->client->sess.sessionTeam)
```